src/Uri.php

```diff
@@ -690,7 +701,13 @@ private function validateState()
                 throw new \InvalidArgumentException('A relative URI must not have a path beginning with a segment containing a colon');
             }
         } elseif (isset($this->path[0]) && $this->path[0] !== '/') {
-            throw new \InvalidArgumentException('The path of a URI with an authority must start with a slash "/" or be empty');
+            @trigger_error(
```

Comment:
This is an accepted way if triggering deprecation errors I think. For library developers I think the `@deprecated` phpdoc is more useful when using a proper IDE. Of course if someone doesn't use something similar to the PHPUnit bridge or use IDEs these days or doesn't read CHANGELOG, then we can't really do anything about it. Also, semver is here to help: when upgrading to a new major version, tests will eventually break, so sooner or later the developer will be notified about the deprecation without breaking actual, stable code.